editor/src/core/shared/math-utils.ts

```diff
@@ -263,6 +263,24 @@ export function sizeFitsInTarget(sizeToCheck: Size, target: Size): boolean {
   return sizeToCheck.width <= target.width && sizeToCheck.height <= target.height
 }
 
+export function rectanglesOverlap(a: CanvasRectangle, b: CanvasRectangle): boolean {
+  const xOverlap = a.x < b.x + b.width && a.x + a.width > b.x
+  const yOverlap = a.y < b.y + b.height && a.y + a.height > b.y
+  return xOverlap && yOverlap
+}
+
+export function rectangleContainsRectangle(
+  outer: CanvasRectangle,
+  inner: CanvasRectangle,
+): boolean {
+  return (
+    rectContainsPoint(outer, canvasPoint({ x: inner.x, y: inner.y })) &&
```

Comment:
More a note for myself, I wonder if there's a way of implementing this that doesn't necessitate all these allocations.

Reply:
This is a good point. I think given how many times this calculation would be run whilst drawing a selection area, we should avoid these extra allocations by hand rolling the calculation (i.e. `outer.x < inner.x && outer.y < inner.y && ...`)

Reply:
good point - fixed here! https://github.com/concrete-utopia/utopia/pull/3750/commits/497ce98368a5b5f21529ffc55175d5aa09288a3d